web-frontend/src/pages/TransferCoins/SendCoins/SendForm.tsx

```diff
@@ -153,6 +179,25 @@ export function SendForm(props: SendFormProps) {
     <Spinner size={12} customClass="inline-block" />
   );
 
+  const handleRecipientChange = useCallback(
+    async (e: ChangeEvent<HTMLInputElement>) => {
+      const value = e.target.value;
+      setRecipient(value);
+      setError({ recipient: '', amount: error?.amount });
+      if (value !== '' && mnsExtensionRegex.test(value)) {
+        const inputMns = value.replace(mnsExtension, '');
+        const targetAddress = await resolveDns(inputMns);
+        if (targetAddress && checkAddressFormat(targetAddress)) {
+          setMnsAddressCorrelation(true);
+        } else {
+          setMnsAddressCorrelation(false);
+          resetTargetMnsAddress();
+        }
+      }
+    },
+    [recipient, resolveDns, resetTargetMnsAddress, mnsExtensionRegex],
```

Comment:
React Hook useCallback has a missing dependency: 'error?.amount'. Either include it or remove the dependency array. You can also replace multiple useState variables with useReducer if 'setError' needs the current value of 'error.amount'